### PR TITLE
fix: support qualified types in topic handlers

### LINKED ISSUE #172: Bug: Pubsub function parameter does not allow additional colons
Let us assume that I have defined the following pubsub function, where the input parameter is of type `serde_json::Value`
```rust
#[topic(pub_sub_name = "pubsub", topic = "A")]
async fn handle_a_event(order: serde_json::Value) {
    println!("Topic A - {:#?}", order)
}
```

When I run `cargo build`, I got the following error
```
error: custom attribute panicked
   --> src/main.rs:111:1
    |
111 | #[topic(pub_sub_name = "rg-pubsub", topic = "A")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: message: assertion `left == right` failed: Expected to only have one input variable
              left: 2
             right: 1
```

However, if I import serde_json::Value ahead of the time and use `Value` instead, no errors will be thrown
```rust
use serde_json::Value;
#[topic(pub_sub_name = "pubsub", topic = "A")]
async fn handle_a_event(order: Value) {
    println!("Topic A - {:#?}", order)
}
```

I believe this might be due to how we are parsing the input variable in the macros [here](https://github.com/ruokun-niu/rust-sdk/blob/master/macros/src/lib.rs#L174)?

## Expected Behavior

The macro is only checking to see if there is only one input parameter. We should not be getting any errors if we use `serde_json::Value` instead of `Value`

## Actual Behavior

Listed above

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by immanuwell**
# Description

`#[topic]` blows up if the handler arg uses a qualified type like `serde_json::Value`.

The macro was splitting tokens on `:` so `serde_json::Value` got treated like 2 inputs. This switches it to parse the function signature with `syn`, keeps the `String` fast path, and adds regression tests. tiny fix, but pretty real.

Repro:
```rust
#[topic(pub_sub_name = "pubsub", topic = "A")]
async fn handle_a_event(order: serde_json::Value) {}
```

Before:
`cargo check` panics with `Expected to only have one input variable`

After:
`cargo check` passes.

## Issue reference

This PR will close #172

## Checklist

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: dapr-macros/src/lib.rs [hunk 1]
```diff
@@ -236,3 +212,70 @@ pub fn topic(args: TokenStream, input: TokenStream) -> TokenStream {
...
+    }
+}
+
+fn is_string_type(ty: &Type) -> bool {
+    match ty {
+        Type::Path(type_path) if type_path.qself.is_none() => type_path
+            .path
+            .segments
+            .last()
+            .is_some_and(|segment| segment.ident == "String"),
```
Comment (mikeee) [inline dapr-macros/src/lib.rs:248]: Was this intentionally added to allow `String` types defined by any other crates such as `other_crate::String`?

File: dapr-macros/src/lib.rs [hunk 2]
```diff
@@ -236,3 +212,70 @@ pub fn topic(args: TokenStream, input: TokenStream) -> TokenStream {
...
+        _ => false,
+    }
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (mikeee) [inline dapr-macros/src/lib.rs:254]: Thank you for adding tests, please enhance this with negative tests to confirm the above review assumptions.

**Review by mikeee — CHANGES_REQUESTED** [review]
I'm in favour of this change with a few changes, please could you clarify whether allowing non-stdlib definitions of `String` should be allowed as implemented?
I think for the string type assertion it'd be better if we asserted against the entire type path, something like this:

```rust
fn is_string_type(ty: &Type) -> bool {
    let Type::Path(type_path) = ty else {
        return false;
    };
    if type_path.qself.is_some() {
        return false;
    }

    let path = &type_path.path;
    let matches_path = |names: &[&str]| {
        path.segments.len() == names.len()
            && path
                .segments
                .iter()
                .zip(names)
                .all(|(seg, name)| seg.ident == *name && seg.arguments.is_empty())
    };

    matches_path(&["String"])
        || matches_path(&["std", "string", "String"])
        || matches_path(&["alloc", "string", "String"])
}
```

wdyt?
